Sprint 04

AntonLPf opened the pull request:
Вроде всё получилось, но пришлось дизейблить кнопки на время обработки ответа пользователя что бы он не мог много раз асинхронно  запустить логику обработки ответа

Commit:
added functionality of 4th stream

DanilaG commented on MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -1,3 +1,10 @@
 
-extension UIColor { }
+extension UIColor {
+    static var ypGreen: UIColor { UIColor(named: "YP Green") ?? UIColor.green }
+    static var ypRed: UIColor { UIColor(named: "YP Red") ?? UIColor.red }
+    static var ypBlack: UIColor { UIColor(named: "YP Black") ?? UIColor.black}
+    static var ypBackground: UIColor { UIColor(named: "YP Background") ?? UIColor.darkGray }
+    static var ypGray: UIColor { UIColor(named: "YP Gray") ?? UIColor.gray }
+    static var ypWhite: UIColor { UIColor(named: "YP White") ?? UIColor.white}
+}
```
Классно, что все цвета вынесены в отдельные константы) Это снижает вероятность опечаток и упрощает ввод цветов)

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,274 @@
+        
+        enableButtons()
+        
+        func getResultViewModel(from resultRecord: ResultsRecord) -> QuizResultsViewModel {
```
Давай вынесим функцию из функции?) Так как это может затруднять чтение кода

AntonLPf replied:
сделал

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,274 @@
+        }
+    }
+    
+    struct QuizStepViewModel {
+        let image: UIImage
+        let question: String
+        let questionNumber: String
+    }
+    
+    struct QuizQuestion {
+        let image: String
+        let text: String
+        let correctAnswer: Bool
+    }
+    
+    struct QuizResultsViewModel {
+      let title: String
+      let text: String
+      let buttonText: String
+    }
+    
+    struct ResultsRecord {
+        let numberOfCorrectAnswers: Int
+        let numberOfqQuestions: Int
+        let date: Date
+        
+        var accuracy: Float {
+            Float(numberOfCorrectAnswers) / Float(numberOfqQuestions) * 100
+        }
+    }
```
🔥, что сразу завёл структурки, они понадобятся в дальнейшем)

DanilaG commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,274 @@
+            
+            let averageAccuracy = logger.getAverageAccuracy()
+            let accuracyString = getPercentageString(for: averageAccuracy)
+            text += "\nСредняя точность: \(accuracyString)%"
```
Классно, что делаешь задание чуть чуть на переёд. Это упростит работу в будующем)

Commit:
fixing number of quaestion bug
fixing image corner radius bug
move functions outside functions
set status bar color to light
